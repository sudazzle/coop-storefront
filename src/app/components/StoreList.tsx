import React from 'react';
import { Address } from '@/app/icons/Address';
import { Phone } from '@/app/icons/Phone';
import { Store } from '@/app/types';

export const StoreOpenStatus: React.FC<{ open: boolean }> = ({ open }) => {
  return (
    <span className={
      `text-sm rounded-md py-1 px-3 ${open ? 'bg-green-400 text-white' : 'bg-red-500 text-red-600'}`
    }>
      {open ? 'Ã…pent' : 'Stengt'}
    </span>
  );
}

export const StoreCard: React.FC<{ store: Store }> = ({ store }) => {
  return (
    <li className="w-[90%] sm:w-[45%] md:w-[32%] rounded-2xl block p-5 bg-[#EFEFE9]">
      <h3 className="font-medium mb-3">{store.Chain}, {store.City}</h3>
      <p className="text-sm gap-1 text-gray-500 flex items-center">
        <span><Address /></span>
        <span>{ store.Address }</span>
      </p>
      <p className="text-sm gap-1 text-gray-500 flex items-center">
        <span><Phone /></span>
        <span>{ store.Phone }</span>
      </p>
      <p className="mt-5 text-sm">
        <StoreOpenStatus open={store.OpenNow} />
        <span className="pl-4 text-gray-500">{store.OpeningHoursToday}</span>
      </p>
    </li>
  );
}

export const StoreList: React.FC<{ stores: Store[] }> = ({ stores }) => {
  return (
    <ul className="flex mt-5 justify-center flex-wrap gap-4">
      {stores.map((store) => (
        <StoreCard key={store.StoreId} store={store} />
      ))}
    </ul>
  );
}