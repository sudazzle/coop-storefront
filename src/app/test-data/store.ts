import { Store } from '@/app/types';

export const stores: Store[] = [
  {
    StoreId: '1',
    Chain: 'Store Chain 1',
    City: 'City 1',
    Address: 'Address 1',
    Phone: '22222222',
    OpenNow: true,
    OpeningHoursToday: '9:00 AM - 5:00 PM',
    Name: 'Store Name 1',
    ChainClassName: 'chainClassName1',
    ChainId: 'chainId1',
    IsEcommerce: true,
    NewspaperUrl: '/newspaper-url-1',
    ChainImage: '/images/chain-image-1.jpg',
    InStoreServices: ['service1', 'service2'],
    Lat: 10.4343434,
    Lng: 10.4343434,
    Distance: 10,
    Email: 'test@email.com',
    OrganizationNumber: '123456789',
    SLag: 'SLag',
    SpecialOpeningHours: [],
    OpeningHours: [],
    AdditionalInformation: '',
  },
  {
    StoreId: '2',
    Chain: 'Store Chain 2',
    City: 'City 2',
    Address: 'Address 2',
    Phone: '12345678',
    OpenNow: true,
    OpeningHoursToday: '9:00 AM - 5:00 PM',
    Name: 'Store Name 2',
    ChainClassName: 'chainClassName2',
    ChainId: 'chainId2',
    IsEcommerce: true,
    NewspaperUrl: '/newspaper-url-2',
    ChainImage: '/images/chain-image-2.jpg',
    InStoreServices: ['service1', 'service2'],
    Lat: 20.4343434,
    Lng: 20.4343434,
    Distance: 20,
    Email: 'test2@email.com',
    OrganizationNumber: '3333333333',
    SLag: 'SLag',
    SpecialOpeningHours: [],
    OpeningHours: [],
    AdditionalInformation: '',
  }
];
