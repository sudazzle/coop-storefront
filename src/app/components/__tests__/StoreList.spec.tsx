import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { StoreList, StoreCard, StoreOpenStatus } from '../StoreList';
import { stores } from '@/app/test-data/store';

describe('StoreCard', () => {
  it('renders a store card', () => {
    const store = stores[0];
    const { container } = render(<StoreCard store={store} />);

    expect(container).toMatchSnapshot();
    expect(screen.getByText(`${store.Chain}, ${store.City}`)).toBeInTheDocument();
    expect(screen.getByText(store.Address)).toBeInTheDocument();
    expect(screen.getByText(store.Phone)).toBeInTheDocument();
    expect(screen.getByText(store.OpeningHoursToday)).toBeInTheDocument();
  });
});

describe('StoreOpenStatus', () => {
  it('renders open status', () => {
    const { container } = render(<StoreOpenStatus open={true} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText('Ã…pent')).toBeInTheDocument();
  });

  it('renders closed status', () => {
    const { container } = render(<StoreOpenStatus open={false} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText('Stengt')).toBeInTheDocument();
  });
});

describe('StoreList', () => {
  it('renders a list of stores', () => {
    const { container } = render(<StoreList stores={stores} />);
    expect(container).toMatchSnapshot();
    expect(screen.getAllByRole('listitem').length).toBe(stores.length);
  });
});
