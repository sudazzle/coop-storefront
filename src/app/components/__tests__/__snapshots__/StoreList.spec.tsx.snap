// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreCard renders a store card 1`] = `
<div>
  <li
    class="w-[90%] sm:w-[45%] md:w-[32%] rounded-2xl block p-5 bg-[#EFEFE9]"
  >
    <h3
      class="font-medium mb-3"
    >
      Store Chain 1
      , 
      City 1
    </h3>
    <p
      class="text-sm gap-1 text-gray-500 flex items-center"
    >
      <span>
        <svg
          fill="gray"
          height="15"
          viewBox="0 0 24 24"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 0c-4.198 0-8 3.403-8 7.602 0 4.198 3.469 9.21 8 16.398 4.531-7.188 8-12.2 8-16.398 0-4.199-3.801-7.602-8-7.602zm0 11c-1.657 0-3-1.343-3-3s1.343-3 3-3 3 1.343 3 3-1.343 3-3 3z"
          />
        </svg>
      </span>
      <span>
        Address 1
      </span>
    </p>
    <p
      class="text-sm gap-1 text-gray-500 flex items-center"
    >
      <span>
        <svg
          fill="gray"
          height="13"
          viewBox="0 0 24 24"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 22.621l-3.521-6.795c-.008.004-1.974.97-2.064 1.011-2.24 1.086-6.799-7.82-4.609-8.994l2.083-1.026-3.493-6.817-2.106 1.039c-7.202 3.755 4.233 25.982 11.6 22.615.121-.055 2.102-1.029 2.11-1.033z"
          />
        </svg>
      </span>
      <span>
        22222222
      </span>
    </p>
    <p
      class="mt-5 text-sm"
    >
      <span
        class="text-sm rounded-md py-1 px-3 bg-green-400 text-white"
      >
        Åpent
      </span>
      <span
        class="pl-4 text-gray-500"
      >
        9:00 AM - 5:00 PM
      </span>
    </p>
  </li>
</div>
`;

exports[`StoreList renders a list of stores 1`] = `
<div>
  <ul
    class="flex mt-5 justify-center flex-wrap gap-4"
  >
    <li
      class="w-[90%] sm:w-[45%] md:w-[32%] rounded-2xl block p-5 bg-[#EFEFE9]"
    >
      <h3
        class="font-medium mb-3"
      >
        Store Chain 1
        , 
        City 1
      </h3>
      <p
        class="text-sm gap-1 text-gray-500 flex items-center"
      >
        <span>
          <svg
            fill="gray"
            height="15"
            viewBox="0 0 24 24"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0c-4.198 0-8 3.403-8 7.602 0 4.198 3.469 9.21 8 16.398 4.531-7.188 8-12.2 8-16.398 0-4.199-3.801-7.602-8-7.602zm0 11c-1.657 0-3-1.343-3-3s1.343-3 3-3 3 1.343 3 3-1.343 3-3 3z"
            />
          </svg>
        </span>
        <span>
          Address 1
        </span>
      </p>
      <p
        class="text-sm gap-1 text-gray-500 flex items-center"
      >
        <span>
          <svg
            fill="gray"
            height="13"
            viewBox="0 0 24 24"
            width="13"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 22.621l-3.521-6.795c-.008.004-1.974.97-2.064 1.011-2.24 1.086-6.799-7.82-4.609-8.994l2.083-1.026-3.493-6.817-2.106 1.039c-7.202 3.755 4.233 25.982 11.6 22.615.121-.055 2.102-1.029 2.11-1.033z"
            />
          </svg>
        </span>
        <span>
          22222222
        </span>
      </p>
      <p
        class="mt-5 text-sm"
      >
        <span
          class="text-sm rounded-md py-1 px-3 bg-green-400 text-white"
        >
          Åpent
        </span>
        <span
          class="pl-4 text-gray-500"
        >
          9:00 AM - 5:00 PM
        </span>
      </p>
    </li>
    <li
      class="w-[90%] sm:w-[45%] md:w-[32%] rounded-2xl block p-5 bg-[#EFEFE9]"
    >
      <h3
        class="font-medium mb-3"
      >
        Store Chain 2
        , 
        City 2
      </h3>
      <p
        class="text-sm gap-1 text-gray-500 flex items-center"
      >
        <span>
          <svg
            fill="gray"
            height="15"
            viewBox="0 0 24 24"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0c-4.198 0-8 3.403-8 7.602 0 4.198 3.469 9.21 8 16.398 4.531-7.188 8-12.2 8-16.398 0-4.199-3.801-7.602-8-7.602zm0 11c-1.657 0-3-1.343-3-3s1.343-3 3-3 3 1.343 3 3-1.343 3-3 3z"
            />
          </svg>
        </span>
        <span>
          Address 2
        </span>
      </p>
      <p
        class="text-sm gap-1 text-gray-500 flex items-center"
      >
        <span>
          <svg
            fill="gray"
            height="13"
            viewBox="0 0 24 24"
            width="13"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 22.621l-3.521-6.795c-.008.004-1.974.97-2.064 1.011-2.24 1.086-6.799-7.82-4.609-8.994l2.083-1.026-3.493-6.817-2.106 1.039c-7.202 3.755 4.233 25.982 11.6 22.615.121-.055 2.102-1.029 2.11-1.033z"
            />
          </svg>
        </span>
        <span>
          12345678
        </span>
      </p>
      <p
        class="mt-5 text-sm"
      >
        <span
          class="text-sm rounded-md py-1 px-3 bg-green-400 text-white"
        >
          Åpent
        </span>
        <span
          class="pl-4 text-gray-500"
        >
          9:00 AM - 5:00 PM
        </span>
      </p>
    </li>
  </ul>
</div>
`;

exports[`StoreOpenStatus renders closed status 1`] = `
<div>
  <span
    class="text-sm rounded-md py-1 px-3 bg-red-500 text-red-600"
  >
    Stengt
  </span>
</div>
`;

exports[`StoreOpenStatus renders open status 1`] = `
<div>
  <span
    class="text-sm rounded-md py-1 px-3 bg-green-400 text-white"
  >
    Åpent
  </span>
</div>
`;
